app/mailers/devise_notify_mailer.rb

```diff
@@ -41,6 +45,11 @@ def confirmation_instructions(record, token, _opts = {})
     )
   end
 
+  def intercept_send?(email)
+    email_domain = email.split("@").last.downcase
```

Comment:
I inverted logic to `Rails.env.staging? && ALLOWLIST.exclude?(email_domain)` and got tests passing and probably easier to read as well